### PR TITLE
DATAES-33 - Polishing

### PR DESCRIPTION
**Pull request by xhaggi**
Improves the implementation of parent property recognition.

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by xhaggi** [comment]
@olivergierke mind taking a look? Only a small polishing to improve the `@Parent` property recognition.

File: src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java [hunk 1]
```diff
@@ -168,18 +168,18 @@ public ElasticsearchPersistentProperty getScoreProperty() {
...
-		}
+			if (parentProperty != null) {
+				throw new MappingException(
+						String.format("Attempt to add parent property %s but already have property %s registered "
+								+ "as version. Check your mapping configuration!", property.getField(), parentProperty.getField()));
+			}
 
-		if (property.isVersionProperty()) {
-			Assert.isTrue(property.getType() == Long.class, "Version property must be of type Long!");
+			Parent parentAnnotation = property.getType().getAnnotation(Parent.class);
```
Comment (olivergierke) [inline src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java]: I'm wondering if that is actually intended here. What the implementation lookups is the `@Parent` annotation here:

```java
@Parent
class SomeType {}

class SomeOtherType {
  SomeType property;
}
```

where previously you would've discovered this:

```
class SomeOtherType {
  @Parent Object property;
}
```

I.e., where previously a property annotation was found we now lookup an annotation on the type of the property. Is that intentional?

Comment (xhaggi) [reply src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java]: You're right. I've fix the issue and force-pushed the branch.

**Comment by xhaggi** [comment]
@olivergierke anything left?

**Comment by xhaggi** [comment]
@olivergierke could you please approve the PR if nothing is left

**Comment by xhaggi** [comment]
@olivergierke I want to merge this in, but your sign-off is missing.

**Pull request closed without merging**